### DISCUSSION TITLE
Wazuh decoder and rules

### DISCUSSION THREAD

**Discussion by tsurara36** [wazuh/wazuh#17517] [Q&A]
Hi all,

I am new to Wazuh, and am trying to create my own custom rule and decoder where it will create an alert on my Wazuh dashboard whenever a .encrypted file has been added to my monitored folder. 
This is my decoder
![image](https://github.com/wazuh/wazuh/assets/131188770/34371710-2bf5-4e16-a082-9c2ab024e712)

This is my rule
![image](https://github.com/wazuh/wazuh/assets/131188770/084b1a2d-631f-4d82-88d6-5d1f4d427aab)

When I run it on Wazuh-Logtest, however this only works after I edit the 0006-json_decoders.xml file, as before this decoder takes priority over my decoder (which I do not understand why).
![image](https://github.com/wazuh/wazuh/assets/131188770/30c7194d-386b-4ccd-8f0a-6abac40121a6)

However, even though I got it to work on Wazuh-Logtest, when I try to add a .encrypted file into my folder, it still uses the old rule 554.
![image](https://github.com/wazuh/wazuh/assets/131188770/9ad00019-4a2a-4ac6-9ef8-613dbec7d34b)

Is there anyways I can overwrite this rule such that my own custom rule will appear on the Wazuh dashboard instead.

Thanks

**Comment by s-ocando** [comment]
Hi @tsurara36,

To detect new `.encrypted` files, you can easily create a custom rule like the following one:

```
<group name="syscheck,">

  <rule id="100021" level="5">
    <if_sid>554</if_sid>
    <field name="file">.encrypted$</field>
    <description>$(file) has been encrypted</description>
  </rule>

</group> 
```

This rule is a child of rule [554 "File added to the system"](https://github.com/wazuh/wazuh/blob/4.4/ruleset/rules/0015-ossec_rules.xml#L227). This means that each time a new file is added to your monitored directory, the analysis engine will verify if its extension is `.encrypted` and if that's the case rule 100021 will be triggered instead of 554.

Add this rule in the `/var/ossec/etc/rules/local_rules.xml` file and restart the Wazuh manager to load the changes: `systemctl restart wazuh-manager`.  Check the [Custom rules and decoders](https://documentation.wazuh.com/current/user-manual/ruleset/custom.html) documentation to learn more. 

Let us know if you have any questions.

**Reply by tsurara36** [reply]
Hi @s-ocando,

I have made the changes to my local_rules.
![image](https://github.com/wazuh/wazuh/assets/131188770/bc61edc4-4a01-4df7-8578-4343ff7337c4)
However, when I run the logtest, I am getting rule 554 instead of my custom rule 100021
![image](https://github.com/wazuh/wazuh/assets/131188770/5ca59036-9ba2-462b-ad64-a842a20a7282)

Thanks

**Reply by s-ocando** [reply]
The syscheck messages are sent through a special channel, and it is not straightforward to test their rules using wazuh-logtest.

I suggest reverting the modifications of the JSON decoder, adding the custom rule I shared earlier and restarting the manager so changes can take effect. Next, create an `.encrypted` file in a [real-time monitored](https://documentation.wazuh.com/current/user-manual/capabilities/file-integrity/basic-settings.html#real-time-monitoring) directory. 

You should see the custom alert, similar to the one shown in the following screenshot:
![image](https://github.com/wazuh/wazuh/assets/61882981/de07772e-08a8-4df1-a1cb-0c5bade236ea)

**Reply by tsurara36** [reply]
Hi @s-ocando,

Thanks for your help, I was able to get my dashboard to display, but I also had to add this to my decoder.
![image](https://github.com/wazuh/wazuh/assets/131188770/96dddf91-e4f0-4067-bfaf-388831d16e09)

**Reply by s-ocando** [reply]
Hi @tsurara36, 

I'm glad to know that you're now receiving the alerts. Nevertheless, I'd like to clarify that if you're using a child rule of rule 554, similar to the one I shared, there's no need to modify any decoder. The creation of a new file in a monitored directory triggers rule 554 and the child rule (100021 in the previous example) checks if the file's name ends with `.encrypted`. There's no need to modify or add any decoder.

**Reply by tsurara36** [reply]
Hi @s-ocando ,

I am using a child rule of 554 like what you showed, however it did not work until I modified my decoder (I do not understand why as my decoder is also not doing anything).

Thanks

**Reply by s-ocando** [reply]
There shouldn't be a need to modify the decoder. If you'd like, feel free to share your ruleset and I'll test it in my environment.

**Reply by tsurara36** [reply]
Hi @s-ocando ,

This is my ruleset
![image](https://github.com/wazuh/wazuh/assets/131188770/d1a8a5a9-7fae-448f-9387-deeb66cdda2c)

**Reply by s-ocando** [reply]
Hi!
I've tested your rule on my environment and it worked as expected without the need to modify any decoder. What decoders have you changed?
Note that your rule uses `<match>encrypted</match>` so it matches if the word "encrypted" is present on any field on the log.  For example, the creation of a file named `encrypted.txt` triggers your custom rule.
I suggest using `<field name="file">.encrypted$</field>` to only search for the extension in the filename.

**Reply by tsurara36** [reply]
Hi @s-ocando,

I am trying to detect the extensions of a file, and I have a list of extensions. 
So when Wazuh detects a file with a different extension, 
I want to check if the extension is part of the list that I have. 

Can this be done?

Thanks

**Reply by s-ocando** [reply]
Hi @tsurara36, 

You can create a custom rule that triggers each time a file is added with an extension not included in the list. For example, 

```
<group name="syscheck,">

  <rule id="100021" level="5">
    <if_sid>554</if_sid>
    <field name="file" negate="yes">.txt$|.png$|.jpg$|.dat$</field>
    <description>$(file) with an extension not allowed has been added</description>
  </rule>

</group>
```
This rules triggers if the new file's extension is not `.txt`, `.png`, `.jpg` or `.dat`. Modify this rule according to your needs and restart your Wazuh manager to load changes.

**Reply by tsurara36** [reply]
Hi @s-ocando,

Thanks for so much for your help.

**Reply by tsurara36** [reply]
Hi  @s-ocando,

I am trying to use Wazuh to detect if port 3389 and usb port are enabled. 
How would you do it?

Thanks

**Reply by tsurara36** [reply]
Hi @s-ocando,

For the detection of port 3389, I have added this to my agent
![image](https://github.com/wazuh/wazuh/assets/131188770/06471f2e-fcb0-467a-8124-95c388557b30)
I also added this rule in my wazuh manager
![image](https://github.com/wazuh/wazuh/assets/131188770/013f999e-9b2e-48cb-98e9-f4978e7d6f1f)
This is the output I get when I run the command
![image](https://github.com/wazuh/wazuh/assets/131188770/c4fe80fc-71dd-4a2d-964a-384a770678bf)
This is the output I get when I turn port 3389 off
![image](https://github.com/wazuh/wazuh/assets/131188770/c8d26441-228c-4d94-8b01-9b862ee2b1ae)

I am not sure why I am not receiving any alerts and am also unsure as to where I am suppose to look for these types of alerts.

Thanks

**Reply by s-ocando** [reply]
Hi @tsurara36, 

I think a [check_diff](https://documentation.wazuh.com/current/user-manual/ruleset/ruleset-xml-syntax/rules.html#check-diff) rule is not a good fit for this case, as the command's output is empty when the port is disabled.
Instead, I'd recommend looking for the word "LISTEN" to determine if the port is enabled. You can modify the rule as follows:
```
  <rule id="100023" level="5" ignore="600">
    <if_sid>530</if_sid>
    <match>ossec: output: '3389 port'</match>
    <regex>(LISTEN)</regex>
    <description>RDP port is enabled</description>
  </rule>
```  
Since your command checks every 5 seconds, you can use the [ignore](https://documentation.wazuh.com/current/user-manual/ruleset/ruleset-xml-syntax/rules.html#rule) option in the custom rule to avoid excessive triggering. In the previous example, the rule will only trigger once every 10 minutes.

Remember to restart both your Wazuh manager and Wazuh agent after making changes to the ruleset and configuration. Please let us know if this resolves your issue.

**Comment by UmAirKhAn-IS** [comment]
Hi,
Can anyone help me create decoders and rules for the following row logs:

Dec 11 18:30:57 Storage alarm: <189>2025-12-11 18:30:57 10.1.70.101     27020939 0x200F002A0015 Informational(0): The login of the user (user name mm_user) from the source (127.0.0.1) succeeded.

Dec 11 18:26:42 Storage alarm: <189>2025-12-11 18:26:42 10.1.70.101     27020889 0x200F002A0016 Informational(0): The user (user name admin) failed to login from the source (10.1.70.1).The error code is 0x40403285.
